ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,162 @@
+import json
+import os
+import sys
+from typing import Any
+from typing import Optional
+
+import anthropic
+
+from ddtrace import config
+from ddtrace.contrib.trace_utils import unwrap
+from ddtrace.contrib.trace_utils import with_traced_module
+from ddtrace.contrib.trace_utils import wrap
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.llmobs._integrations import AnthropicIntegration
+from ddtrace.pin import Pin
+
+from .utils import _get_attr
+
+
+log = get_logger(__name__)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(anthropic, "__version__", "")
+
+
+config._add(
+    "anthropic",
+    {
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_ANTHROPIC_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "span_char_limit": int(os.getenv("DD_ANTHROPIC_SPAN_CHAR_LIMIT", 128)),
+    },
+)
+
+
+def _extract_api_key(instance: Any) -> Optional[str]:
+    """
+    Extract and format LLM-provider API key from instance.
+    """
+    client = getattr(instance, "_client", "")
+    if client:
+        return getattr(client, "api_key", None)
+    return None
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    operation_name = func.__name__
+
+    span = integration.trace(
```

Comment:
Using `core.context_with_data` here would loosen the coupling between instrumentation and the LLMObs product. I think that's an important change to make in the near future.

Reply:
We're aiming to do a refactor of the LLM integrations to use core context management instead in Q3/Q4. We'll keep this for now for the sake of deadlines, but just wanted to let you know we're definitely keeping that in mind @emmettbutler